pkg/controller/statusmanager/status_manager.go

```diff
@@ -107,55 +112,130 @@ func (status *StatusManager) deleteRelatedObjectsNotRendered(co *configv1.Cluste
+		oc.Status.Version = os.Getenv("RELEASE_VERSION")
+
+		// If we have no conditions at all, pull them from the ClusterOperator object.
+		if len(oc.Status.Conditions) == 0 {
+			co := &configv1.ClusterOperator{ObjectMeta: metav1.ObjectMeta{Name: status.name}}
+			err := status.client.Get(context.TODO(), types.NamespacedName{Name: status.name}, co)
+			if err != nil {
+				log.Printf("failed to retrieve ClusterOperator object: %v - continuing", err)
+			}
+
+			for _, condition := range co.Status.Conditions {
+				v1helpers.SetOperatorCondition(&oc.Status.Conditions, operv1.OperatorCondition{
+					Type:               string(condition.Type),
+					Status:             operv1.ConditionStatus(condition.Status),
+					LastTransitionTime: condition.LastTransitionTime,
+					Reason:             condition.Reason,
+					Message:            condition.Message,
+				})
 			}
 		}
+
 		for _, condition := range conditions {
-			v1helpers.SetStatusCondition(&co.Status.Conditions, condition)
+			v1helpers.SetOperatorCondition(&oc.Status.Conditions, condition)
 		}
 
-		progressingCondition := v1helpers.FindStatusCondition(co.Status.Conditions, configv1.OperatorProgressing)
-		availableCondition := v1helpers.FindStatusCondition(co.Status.Conditions, configv1.OperatorAvailable)
-		if availableCondition == nil && progressingCondition != nil && progressingCondition.Status == configv1.ConditionTrue {
-			v1helpers.SetStatusCondition(&co.Status.Conditions,
-				configv1.ClusterOperatorStatusCondition{
-					Type:    configv1.OperatorAvailable,
-					Status:  configv1.ConditionFalse,
+		progressingCondition := v1helpers.FindOperatorCondition(oc.Status.Conditions, operv1.OperatorStatusTypeProgressing)
+		availableCondition := v1helpers.FindOperatorCondition(oc.Status.Conditions, operv1.OperatorStatusTypeAvailable)
+		if availableCondition == nil && progressingCondition != nil && progressingCondition.Status == operv1.ConditionTrue {
+			v1helpers.SetOperatorCondition(&oc.Status.Conditions,
+				operv1.OperatorCondition{
+					Type:    operv1.OperatorStatusTypeAvailable,
+					Status:  operv1.ConditionFalse,
 					Reason:  "Startup",
 					Message: "The network is starting up",
 				},
 			)
 		}
 
-		v1helpers.SetStatusCondition(&co.Status.Conditions,
-			configv1.ClusterOperatorStatusCondition{
-				Type:   configv1.OperatorUpgradeable,
-				Status: configv1.ConditionTrue,
+		v1helpers.SetOperatorCondition(&oc.Status.Conditions,
+			operv1.OperatorCondition{
+				Type:   operv1.OperatorStatusTypeUpgradeable,
+				Status: operv1.ConditionTrue,
 			},
 		)
 
+		operStatus = &oc.Status
+
+		if equality.Semantic.DeepEqual(oc.Status, oldStatus) {
+			log.Printf("shortcut")
+			return nil
+		}
+
+		buf, err := yaml.Marshal(oc.Status.Conditions)
+		if err != nil {
+			buf = []byte(fmt.Sprintf("(failed to convert to YAML: %s)", err))
+		}
+
+		if err := status.client.Update(context.TODO(), oc); err != nil {
+			if errors.IsNotFound(err) { // not used except for in unit tests
```

Comment:
if it never happens in reality shouldn't we stop unit testing it?

Reply:
yes. fixed.